### DISCUSSION TITLE
How to mirror a Rust struct which contains SecStr?

### DISCUSSION THREAD

**Discussion by chenghang6** [fzyzcjy/flutter_rust_bridge#2314] [Q&A] [answered]
Hi, is it possible to mirror a rust struct like this?
```
pub enum Value { 
    Bytes(Vec<u8>),
    Unprotected(String),
    Protected(SecStr),
}
```
If it's Protected, I want to pass a masked string "********" to Dart instead of the real value.

**Comment by fzyzcjy** [comment, marked as answer]
What about writing a https://cjycode.com/flutter_rust_bridge/guides/types/translatable/custom

**Reply by chenghang6** [reply]
> What about writing a https://cjycode.com/flutter_rust_bridge/guides/types/translatable/custom

I can't make it work. Is there a complete example? Thanks.

**Reply by fzyzcjy** [reply]
Yes, https://github.com/fzyzcjy/flutter_rust_bridge/blob/master/frb_example/pure_dart/rust/src/api/custom_ser_des.rs

**Comment by chenghang6** [comment]
Thanks! @fzyzcjy .

```
#[flutter_rust_bridge::frb(rust2dart(dart_type = "Value", dart_code = "Value.letsParseIt({})"))]
pub fn encode_value(raw: Value) -> String {
    "".to_owned()
}

#[flutter_rust_bridge::frb(dart2rust(dart_type = "Value", dart_code = "{}.letsEncodeIt()"))]
pub fn decode_value(raw: String) -> Value {
    Value::Bytes(vec![])
}

#[flutter_rust_bridge::frb(mirror(CustomDataItem))]
pub struct _CustomDataItem {
    pub value: Option<Value>,
    pub last_modification_time: Option<NaiveDateTime>,
}
```
Will "flutter_rust_bridge_codegen generate" generate a Dart class "Value"?
Got a compile error:
```
  SEVERE: error[E0412]: cannot find type `Value` in this scope
  SEVERE: error[E0412]: cannot find type `Value` in this scope
  SEVERE: error[E0599]: the method `into_into_dart` exists for enum `Option<Value>`, but its trait bounds were not satisfied
  SEVERE: error[E0599]: the method `into_into_dart` exists for struct `HashMap<String, Value>`, but its trait bounds were not satisfied
```

**Reply by fzyzcjy** [reply]
> Will "flutter_rust_bridge_codegen generate" generate a Dart class "Value"?

You can have a try, IIRC it may not do so.

> cannot find type `Value` in this scope

Try to `pub use path::to::your::Value`
